service/dps/storm/io/src/main/java/eu/europeana/cloud/service/dps/storm/io/ReadDatasetBolt.java

```diff
@@ -85,12 +80,15 @@ public void execute(StormTaskTuple t) {
 
     @Override
     public void prepare() {
-        datasetClient = new DataSetServiceClient(ecloudMcsAddress, username, password);
-        fileClient = new FileServiceClient(ecloudMcsAddress, username, password);
+        datasetClient = new DataSetServiceClient(ecloudMcsAddress);
+        fileClient = new FileServiceClient(ecloudMcsAddress);
     }
 
     private void emitFilesFromDataSets(StormTaskTuple t, List<String> dataSets) {
         String representationName = t.getParameter(PluginParameterKeys.REPRESENTATION_NAME);
+        String authorizationHeader = t.getParameter(PluginParameterKeys.AUTHORIZATION_HEADER);
+        fileClient = fileClient.useAuthorizationHeader(authorizationHeader);
```

Comment:
It's assign the same object. New object will not be created. FileClient could be refactored into:

    public void useAuthorizationHeader(final String headerValue){
        client.register(new ECloudBasicAuthFilter(headerValue));
        return;
    }